### ISSUE TITLE
Investigate/Brainstorming Queries/Crons/Mutations

### ISSUE THREAD

**Issue by gmbronco** [balancer/backend#252]
Opening a ticket to start a discussion around improving API's DX. Currently we are implementing different interfaces query-by-query. There are some standards we could adopt. For example pagination would be a good start. A lot of queries seem to be redundant as well.

To move forward, I propose we:

* Discuss if we need this
* Audit queries to identify areas to cleanup and standardise
* Review best practices with the FE team
* Implement refactoring with backward versioning

I believe we can improve our API coherence, efficiency, and DX. I'll appreciate your thoughts and insights.

**Comment by franzns** [comment]
Agree! #137 is also related to this. I believe there are a ton of unused queries but havent yet found a way to actually find them. Stellate isnt working properly for this unfortunately.

**Comment by franzns** [comment]
We should standardise some things on the crons and mutations/tasks asap:
- Crons: 
  - Take `Chain[]`(not ID) as input, but can still be called using ID in the HTTP request 
  - Crons should always be per chain and per version to limit blast radius in case of failure. It makes sense in some cases to have multichain crons
  - Should define a good structure for multichain crons instead of just adding them to the mainnet network config.
- Mutations/Tasks
  - Similar to crons, mutations should take a `Chain[]` argument
  - Have a return object that defines which chains/subtasks worked and which failed

We need to discuss and brainstorm about how to standardise and "align" queries, gql schema and db schema:
Some Ideas:
- Queries: 
  -  always use `where:{...}` to filter
  - If no chain is given, results for all chains are returned
  - use paging by default `first`, `skip`, `limit` , introduce proper default and max limit (1k?)
  - use `orderBy` and `orderDirection`
  - Take any-case input but we always return lower-case (#307 )

- We should also align the DB schema and make sure we only query in the DB what was queried in gql. 
- We also want to think about separating fields in the schemas between mutable and immutable data for easier documentation/querying. 
- For queries we should also start to introduce per-field caching with annotations to make the most out of Stellate
